fjpgtt/pollitos-league-manager
# Sports League Management System Implementation

Sports League Management System Implementation

Changes made:
- Creation of project base structure with models (League, Team, Player, Match)
- Implementation of DTOs to maintain a clear separation between API and business logic
- Profile configuration:
  * default: Initialization with empty leagues
  * populated: Data preload with 2 teams and 2 players per team
- Implementation of endpoints for:
  * Team management (create, edit, list)
  * Player management (create, edit, list, delete)
  * Match management (create, list, delete)
- Architecture improvements:
  * Use of records for DTOs
  * Log implementation
  * Error handling
  * Business validations

The implementation includes all requested functionalities and follows Spring Boot best practices.

Steps to test:
1. Run with default profile to verify empty initialization
2. Run with populated profile to verify data preload
